File: Day277_2025/Q277_Leetcode.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        // Q. Container With Most Water
        int left = 0, right = height.size() - 1, maxArea = 0;
        while(left < right){
            int b = right - left;
            int l = height[left] < height[right] ? height[left++] : height[right--];
            maxArea = max(maxArea, l * b);
        }

        return maxArea;
    }
};

File: Day277_2025/Q277_Leetcode.java
class Q277_GFG {
    public int maxArea(int[] height) {
        int left = 0, right = height.length - 1;
        int maxArea = 0;

        while (left < right) {
            int width = right - left;
            int h = Math.min(height[left], height[right]);
            maxArea = Math.max(maxArea, h * width);

            // Move the smaller height pointer inward
            if (height[left] < height[right]) {
                left++;
            } else {
                right--;
            }
        }

        return maxArea;
    }
}
